contrib/agent/src/main/java/io/opencensus/contrib/agent/ContextStrategyImpl.java

```diff
@@ -22,8 +24,27 @@
  */
 final class ContextStrategyImpl implements ContextStrategy {
 
+  private final Cache<Thread, Context> savedContexts
+          = CacheBuilder.newBuilder().weakKeys().build();
```

Comment:
So, iiuc, you use weakKeys() so that threads are not kept around when they're done. Is there any danger of new thread being created that reuses a thread id and causes a false positive here?